src/ports/postgres/modules/deep_learning/test/unit_tests/test_madlib_keras.py_in

```diff
@@ -803,6 +810,267 @@ class MadlibKerasHelperTestCase(unittest.TestCase):
+
+        starting_image_count = 2*len(self.dependent_var)
+        ending_image_count = starting_image_count + len(self.dependent_var)
+        k = {'SD' : {}}
+
+        model_state = [self.loss, self.accuracy, starting_image_count]
+        model_state.extend(self.model_weights)
+        model_state = np.array(model_state, dtype=np.float32)
+
+        self.subject.compile_and_set_weights(self.model, self.compile_params,
+                                             '/cpu:0', model_state.tostring(), self.model_shapes)
+
+        state = [self.loss * starting_image_count, self.accuracy * starting_image_count, starting_image_count]
+
+        k['SD']['segment_model'] = self.model
+        new_state = self.subject.internal_keras_eval_transition(
+            state, self.dependent_var , self.independent_var, self.model.to_json(), 'dummy_model_data',
+            None, 0, 3, self.all_seg_ids, self.total_images_per_seg,
+            0, **k)
+
+        agg_loss, agg_accuracy, image_count = new_state
+
+        self.assertEqual(ending_image_count, image_count)
+        # set_session is only called in first buffer, not here
+        self.assertEqual(0, self.subject.K.set_session.call_count)
+        # loss and accuracy should be unchanged
+        self.assertAlmostEqual(self.loss * ending_image_count, agg_loss, 4)
+        self.assertAlmostEqual(self.accuracy * ending_image_count, agg_accuracy, 4)
+        # Clear session and sess.close must get called for the last buffer in gpdb,
+        #  but not in postgres
+        self.assertEqual(0 if is_platform_pg else 1, self.subject.clear_keras_session.call_count)
+
+    def test_internal_keras_eval_transition_first_buffer_pg(self):
+        self._test_internal_keras_eval_transition_first_buffer(True)
+
+    def test_internal_keras_eval_transition_first_buffer_gpdb(self):
+        self._test_internal_keras_eval_transition_first_buffer(False)
+
+    def test_internal_keras_eval_transition_middle_buffer_pg(self):
+        self._test_internal_keras_eval_transition_middle_buffer(True)
+
+    def test_internal_keras_eval_transition_middle_buffer_gpdb(self):
+        self._test_internal_keras_eval_transition_middle_buffer(False)
+
+    def test_internal_keras_eval_transition_last_buffer_pg(self):
+        self._test_internal_keras_eval_transition_last_buffer(True)
+
+    def test_internal_keras_eval_transition_last_buffer_gpdb(self):
+        self._test_internal_keras_eval_transition_last_buffer(False)
+
+    def test_internal_keras_eval_merge(self):
+        image_count = self.total_images_per_seg[0]
+        state1 = [3.0*self.loss, 3.0*self.accuracy, image_count]
+        state1 = state1
+        state2 = [2.0*self.loss, 2.0*self.accuracy, image_count+30]
+        state2 = state2
+        merged_state = self.subject.internal_keras_eval_merge(state1,state2)
+        state = merged_state
+        agg_loss = state[0]
+        agg_accuracy = state[1]
+        image_count_total = state[2]
```

Comment:
We can directly use `merged_state` instead of `state`, and remove the `state` variable.